webapp/src/js/controllers/contacts-edit.js

```diff
@@ -62,7 +62,7 @@ angular.module('inboxControllers').controller('ContactsEditCtrl',
     });
 
     const getFormInstanceData = function() {
-      const type = ctrl.contact && (ctrl.contact.contact_type || ctrl.contact.type);
+      const type = ctrl.contact && ContactTypes.getTypeId(ctrl.contact);
```

Comment:
Minor: `getTypeId` does a null check for you.

```suggestion
      const type = ContactTypes.getTypeId(ctrl.contact);
```